app/lib/actions/katello/repository/upload_files.rb

```diff
@@ -6,9 +6,10 @@ module Actions
   module Katello
     module Repository
       class UploadFiles < Actions::EntryAction
-        def plan(repository, files)
+        def plan(repository, files, content_type = nil)
           action_subject(repository)
           tmp_files = prepare_tmp_files(files)
+          unit_type_id = (content_type ? SmartProxy.pulp_master.content_service(content_type)::CONTENT_TYPE : repository.unit_type_id)
```

Comment:
same comment here, use default_managed_content_type from the repository_type manager